app/models/ext_management_system.rb

```diff
@@ -426,6 +426,24 @@ def enable!
     update!(:enabled => true)
   end
 
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    workers = MiqWorker.find_current_or_starting.where(:queue_name => "ems_#{id}") #better way to get 'queue_name'?
+    if workers.count.zero?
+      destroy_queue # from AsyncDeleteMixin
+    else
+      # Something like a 15 seconds before trying again?
+      MiqQueue.put(
+          :deliver_on  => 15.seconds.from_now,
+          :class_name  => self.class.name,
+          :instance_id => id,
+          :method_name => "orchestrate_destroy"
```

Comment:
I have a silly question, since we recently added the ability to disable a provider, can we prevent deletion of enabled providers?  The disable could be responsible for making sure any ems workers are also stopped before it's marked at disable.  In other words, force them to first disable the provider, then delete it?